src/aiohomeconnect/client.py

```diff
@@ -676,21 +703,36 @@ async def get_all_events(
         * [Program progress changes](https://api-docs.home-connect.com/events#program-progress-changes)
         * [Home appliance state changes](https://api-docs.home-connect.com/events#home-appliance-state-changes)
         """
-        response = await self._auth.request(
+        # We use 60 seconds timeout because at least every 55 seconds a KEEP-ALIVE event
+        # will be sent. See https://api-docs.home-connect.com/events/#availability-matrix
+        async with self._auth.connect_sse(
             "GET",
             "/homeappliances/events",
-            headers={"Accept-Language": accept_language},
-        )
-        return ArrayOfEvents.from_dict(response.json()["data"])
-
-    async def get_events(
+            timeout=Timeout(60),
+            headers={
+                "Accept-Language": accept_language,
+            },
+        ) as event_source:
+            async for sse in event_source.aiter_sse():
+                if sse.event == EventType.KEEP_ALIVE:
+                    continue
+                data = json.loads(sse.data)
+                if "items" in data:
+                    for item in data["items"]:
+                        item["haId"] = data["haId"]
+                        event = Event.from_dict(item)
+                        yield EventMessage.from_server_sent_event(sse, event)
+                else:
+                    event = Event.from_dict(data)
+                    yield EventMessage.from_server_sent_event(sse, data)
```

Comment:
It would be nice to add tests for these two cases, ie with `items` and without. Can you copy the example responses you gathered earlier and use to write a couple of tests for this?